astronomy/frames.hpp

```diff
@@ -19,57 +19,131 @@ using quantities::si::ArcMinute;
-// The frame is the International Celestial Reference Frame.
-// The basis is defined from the orbit of the Earth at J2000.0 as follows:
-// The xy plane is the plane of the Earth's orbit at J2000.0.
-// The x axis is out along the ascending node of the instantaneous plane of the
-// Earth's orbit and the Earth's mean equator at J2000.0.
-// The z axis is perpendicular to the xy-plane in the directional (+ or -) sense
-// of Earth's north pole at J2000.0.
-// The basis is right-handed and orthonormal.
-using ICRFJ2000Ecliptic =
-    Frame<serialization::Frame::SolarSystemTag,
-          serialization::Frame::ICRF_J2000_ECLIPTIC, true>;
+// The International Celestial Reference System.
+// The origin is the barycentre of the solar system.  The axes are fixed with
+// respect to the quasars.
+// The xy plane is close to the mean equator at J2000.0.
+// The x axis is close to the dynamical equinox at J2000.
+// The z axis is perpendicular to the xy plane and points to the northern side
+// of the xy plane.
+// To be honest:
+// - There are several realizations of the ICRS, namely ICRF (ICRF1) and its
+//   extensions, ICRF2, and the optical Hipparcos Celestial Reference Frame
+//   (HCRF); these realizations have different errors, but are consistent. ICRF3
+//   is in preparation, as well as an optical realization based on Gaia data.
+// - The terms Barycentric Celestial Reference System (BCRS) and ICRS are often
+//   used interchangeably; the relativistic framework, including the metric
+//   tensor, is defined for the BCRS (IAU 2000 resolution B1.3); the orientation
+//   of the BCRS is given by the ICRS axes unless otherwise stated (IAU 2006
+//   resolution B2 recommendation 2).
+using ICRS = Frame<serialization::Frame::SolarSystemTag,
+                   serialization::Frame::ICRS,
+                   /*frame_is_inertial=*/true>;
 
-// The xy plane is the plane of the Earth's mean equator at J2000.0.
-// The x axis is out along the ascending node of the instantaneous plane of the
-// Earth's orbit and the Earth's mean equator at J2000.0.
-// The z axis is along the Earth's mean north pole at J2000.0.
-// The basis is right-handed and orthonormal.
-// Note that |ICRFJ2000Equator| and |ICRFJ2000Ecliptic| share their x axis.
-using ICRFJ2000Equator =
-    Frame<serialization::Frame::SolarSystemTag,
-          serialization::Frame::ICRF_J2000_EQUATOR, true>;
+// The Geocentric Celestial Reference System.
+// The origin is the centre of mass of the whole Earth system, including oceans
+// and atmosphere; the axes are those of the ICRS.
+// To be honest:
+// - The BCRS and GCRS are relativistic reference systems.
+// - The transformation between BCRS and GCRS spatial coordinates has no
+//   rotational component, i.e., the GCRS is kinematically non-rotating with
+//   respect to the BCRS; as a result, it rotates dynamically with respect to
+//   the BCRS; in particular, the equations of motion in the GCRS include de
```

Comment:
I don't understand.  Does it rotate or not?

Reply:
http://articles.adsabs.harvard.edu/cgi-bin/nph-iarticle_query?db_key=AST&bibcode=1998A%26A...334.1123K&letter=.&classic=YES&defaultprint=YES&whole_paper=YES&page=1123&epage=1123&send=Send+PDF&filetype=.pdf